File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.rb
# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
  l, r = 0, numbers.length-1
  while l < r
    if numbers[l]+numbers[r] > target
      r -= 1
    elsif numbers[l]+numbers[r] < target
      l += 1
    else
      return [l+1, r+1]
    end
  end
end

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.ts
function twoSum(numbers: number[], target: number): number[] {
  let l = 0
  let r = numbers.length - 1
  while (l < r) {
    if (numbers[l] + numbers[r] > target) {
      r--
    } else if (numbers[l] + numbers[r] < target) {
      l++
    } else {
      return [l + 1, r + 1]
    }
  }
}